src/store/dtree.ts

```diff
@@ -309,6 +327,55 @@ class DtreeStore {
     })
   }
 
+  private updateDtreeCode(
+    updater: () => Promise<void>,
+    errorMessage: string,
+  ): void {
+    this.isDtreeLoading = true
+
+    updater()
+      .catch(() => {
+        showToast(errorMessage, 'error')
+      })
+      .finally(() => {
+        runInAction(() => {
+          this.isDtreeLoading = false
+        })
+      })
+  }
+
+  private loadDtree(dtreeName: string): void {
+    this.updateDtreeCode(
```

Comment:
These two functions look strange. Why break `then` and `catch` onto different methods? Is `updateDtreeCode` supposed to be reusable? But it's private.

Reply:
I thought of` joinDtree` probably, but there is no such function for dtree